specs/spec.md

```diff
@@ -0,0 +1,319 @@
+# @TODO
+- [ ] Link to parsers.
+- [ ] attestations
+
+# Hobbits
+
+*Written by Dean Eigenmann <dean@status.im> & Rene Nayman <rene@whiteblock.io>* 
+
+## Table of Contents
+
+1. [Abstract](#abstract)
+2. [Flow](#flow)
+3. [Wire Protocol](#wire-protocol)
+    3.1 [Messages](#messages)
+4. [Transport](#transport)
+5. [Protocols](#protocols)
+    5.1 [RPC](#rpc)
+    5.2 [Gossip](#gossip)
+    5.3 [PING](#ping)
+6. [Implementations](#implementations)
+
+## Abstract
+
+Hobbits is a modular wire protocol which allows implementers to experiment with various application logic over a practical network in an agnostic and expedient manner.
+
+Hobbits wire protocol was developed so that Eth2.0 clients could begin network-level testing without requiring libp2p.
+
+## Flow
+
+We layout a network where the nodes are peered in the following manner:
+
+![nodes](./nodes.png)
+
+The following flow dictates how blocks are shared amongst peers:
+
+![sequence diagram](./diagram.png)
+
+1. Node 1 receives a new block `OxCAB` and gossips it to Node 2 and Node 3
```

Comment:
```suggestion
1. Node 1 receives a new block `0xCAB` and gossips `hash(0xCAB)` to Node 2 and Node 3
```

Can we make it clear that we are gossiping hashes and not the full block? In this workflow.